src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ClaimantTellSomethingElseHelper.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.helpers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.items.GenericTseApplicationTypeItem;
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.documents.RespondentTellSomethingElseData;
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.documents.RespondentTellSomethingElseDocument;
+import uk.gov.hmcts.ethos.replacement.docmosis.utils.TSEApplicationTypeData;
+
+import java.time.LocalDate;
+
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.CLAIMANT_TITLE;
+import static uk.gov.hmcts.ethos.replacement.docmosis.constants.TSEConstants.CLAIMANT_TSE_WITHDRAW_CLAIM;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.TornadoService.CLAIMANT_TSE_FILE_NAME;
+
+public final class ClaimantTellSomethingElseHelper {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private static final String CLAIMANT_TSE_TEMPLATE_NAME = "EM-TRB-EGW-ENG-02822.docx";
+
+    private ClaimantTellSomethingElseHelper() {
+    }
+
+    public static String getDocumentRequest(CaseData caseData, String accessKey)
+            throws JsonProcessingException {
+
+        TSEApplicationTypeData selectedAppData = getSelectedApplicationType(caseData);
+        GenericTseApplicationTypeItem lastApp = getCurrentGenericTseApplicationTypeItem(caseData);
+
+        RespondentTellSomethingElseData data = RespondentTellSomethingElseData.builder()
```

Comment:
Is is worth refactoring the main class name from RespondentTellSomethingElseData to something more generic like TseApplicationData if we are going to be using it for the claimant side as well?

Reply:
resolved in [3e0062c](https://github.com/hmcts/et-ccd-callbacks/pull/2036/commits/3e0062cc28dc002b093ce6b5a2aae47f42f0295e)